CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java

```diff
@@ -2206,6 +2208,43 @@ private void updateLayerMask(final int attributeId, final GraphElementType eleme
 
     // update the layer visibility for an element when an attribute changes
     private void updateLayerMask(final int attributeId, final int elementid) {
+        if(transactionLayerVisibilityAttributeId < 0) {
```

Comment:
@aldebaran30701 if there isn't a specific reason to do this I would prefer the explicit check for `Graph.NOT_FOUND` which keeps this consistent with the rest of the app.

FYI going off on a tangent and a more for your info, we also have `AttributeUpdateProvider` which can be used to "upgrade" attributes to a new version. Say you stored the layer visibility as a String and you wanted to change it to an List (for whatever reason) you can use an update provider to upgrade the attribute during load.